### PR TITLE
pollymc-unwrapped: init at 8.0

### PR DESCRIPTION
**Pull request by s0me1newithhand7s**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by TheMaxMur — APPROVED** [review]

**Review by Hezaki — APPROVED** [review]

**Review by nakidai — APPROVED** [review]

File: pkgs/by-name/po/pollymc-unwrapped/package.nix [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+) "gamemodeSupport is only available on Linux.";
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "pollymc-unwrapped";
+  version = "8.0";
+
+  src = fetchFromGitHub {
+    owner = "fn2006";
+    repo = "PollyMC";
+    url = "https://github.com/fn2006/PollyMC/releases/download/8.0/PollyMC-8.0.tar.gz";
```
Comment (Aleksanaa) [inline pkgs/by-name/po/pollymc-unwrapped/package.nix:47]:
```suggestion
```

File: pkgs/by-name/po/pollymc-unwrapped/package.nix [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    rm -rf source/libraries/libnbtplusplus
+    ln -s ${libnbtplusplus} source/libraries/libnbtplusplus
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    extra-cmake-modules
+    jdk17
+    stripJavaArchivesHook
```
Comment (Aleksanaa) [inline pkgs/by-name/po/pollymc-unwrapped/package.nix:62]:
```suggestion
    stripJavaArchivesHook
    kdePackages.wrapQtAppsHook
```

File: pkgs/by-name/po/pollymc-unwrapped/package.nix [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    description = "Free, open source launcher for Minecraft";
+    longDescription = ''
+      Allows you to have multiple, separate instances of Minecraft (each with
+      their own mods, texture packs, saves, etc) and helps you manage them and
+      their associated options with a simple interface.
+    '';
+    homepage = "https://github.com/fn2006/PollyMC";
+    changelog = "https://github.com/fn2006/PollyMC/releases/tag/${finalAttrs.src.rev}";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ ];
```
Comment (Aleksanaa) [inline pkgs/by-name/po/pollymc-unwrapped/package.nix:114]: New packages **must** have a maintainer.

File: pkgs/by-name/po/pollymc-unwrapped/package.nix [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      # we wrap our binary manually
+      (lib.cmakeFeature "INSTALL_BUNDLE" "nodeps")
+      # disable built-in updater
+      (lib.cmakeFeature "MACOSX_SPARKLE_UPDATE_FEED_URL" "''")
+      (lib.cmakeFeature "CMAKE_INSTALL_PREFIX" "${placeholder "out"}/Applications/")
+    ];
+
+  dontWrapQtApps = true;
```
Comment (Aleksanaa) [inline pkgs/by-name/po/pollymc-unwrapped/package.nix:98]: (So why we don't wrap it?)

File: pkgs/by-name/po/pollymc-unwrapped/package.nix [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+  msaClientID ? null,
+  gamemodeSupport ? stdenv.isLinux,
+}:
+
+let
+  libnbtplusplus = fetchFromGitHub {
+    owner = "PrismLauncher";
+    repo = "libnbtplusplus";
+    rev = "a5e8fd52b8bf4ab5d5bcc042b2a247867589985f";
```
Comment (Aleksanaa) [inline pkgs/by-name/po/pollymc-unwrapped/package.nix:27]: Is there any hint somewhere upstream which commit they are using?

File: pkgs/by-name/po/pollymc-unwrapped/package.nix (hunk 3 above)
Comment (s0me1newithhand7s) [reply pkgs/by-name/po/pollymc-unwrapped/package.nix:114]: i can add myself? :D

File: pkgs/by-name/po/pollymc-unwrapped/package.nix (hunk 4 above)
Comment (s0me1newithhand7s) [reply pkgs/by-name/po/pollymc-unwrapped/package.nix:98]: as i've take this package.nix from Prism Launcher's one i don't think this question is suitable here, but i may have an idea. as there is two "prismlauncher" binaries, and one of 'em is unwrapped. i also decided to do this.

File: pkgs/by-name/po/pollymc-unwrapped/package.nix (hunk 5 above)
Comment (s0me1newithhand7s) [reply pkgs/by-name/po/pollymc-unwrapped/package.nix:27]: nope, but i suggest - last commit, as i've built both prism launcher and pollymc with this option and got last commit from repo.

File: pkgs/by-name/po/pollymc-unwrapped/package.nix (hunk 1 above)
Comment (s0me1newithhand7s) [reply pkgs/by-name/po/pollymc-unwrapped/package.nix:47]: will this work? on my local machine not including url with repo caused to "404" and wrong url picking.

File: pkgs/by-name/po/pollymc-unwrapped/package.nix (hunk 2 above)
Comment (s0me1newithhand7s) [reply pkgs/by-name/po/pollymc-unwrapped/package.nix:62]: what.. what does it mean?

File: pkgs/by-name/po/pollymc-unwrapped/package.nix (hunk 1 above)
Comment (h7x4) [reply pkgs/by-name/po/pollymc-unwrapped/package.nix:47]: It *should* work. The github fetcher uses the `owner` and `repo` to construct itself a url to fetch from. It shouldn't be necessary to set it. The `rev` looks correct too.

Comment (s0me1newithhand7s) [reply pkgs/by-name/po/pollymc-unwrapped/package.nix:47]: 'k, i'll commit and do force push as soon as i get back to home.

**Comment by s0me1newithhand7s** [comment]
> Not sure if it's a good idea to risk DMCA abuse against nixpkgs, you could just override prismlauncher with relevant patches if you really need this.

reason behind this PR is not just me. i'm trying to help peoples who wanna play minecraft on actually good launcher but struggles with building this package from NUR. if this package against some rules like "no DRM-free minecraft launchers" - sure, i'll close PR, delete commit and find new package to maintain.

**Comment by unabomberlive** [comment]
> Not sure if it's a good idea to risk DMCA abuse against nixpkgs, you could just override prismlauncher with relevant patches if you really need this. 

Fuck the rules - **FUCK DMCA**

File: pkgs/by-name/po/pollymc-unwrapped/package.nix (hunk 5 above)
Comment (s0me1newithhand7s) [reply pkgs/by-name/po/pollymc-unwrapped/package.nix:27]: i'll leave it like this, bc after two more test builds with `PolyMC/libnbtplusplus` and `MultiMC/libnbtplusplus` nothing changed, binary builded and worked just fine

**Comment by s0me1newithhand7s** [comment]
about reasons why this package named "unwrapped":
[here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/pr/prismlauncher/package.nix) we can see how `prismlauncher` isn't seporate package with it's own `fetchFromGit`. as i've tested - this "package" can build wrapped version of **any** `MultiMC` fork. in case my `pollymc-unwrapped` get merged - i'll make wrapped version, just like `PrismLauncher` team did.

**Review by nakidai — APPROVED** [review]

File: pkgs/by-name/po/pollymc-unwrapped/package.nix [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+      their associated options with a simple interface.
+    '';
+    homepage = "https://github.com/fn2006/PollyMC";
+    changelog = "https://github.com/fn2006/PollyMC/releases/tag/${finalAttrs.src.rev}";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [
+        s0me1newithhand7s
+    ];
+    mainProgram = "pollymc";
+    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```
Comment (JohnRTitor) [inline pkgs/by-name/po/pollymc-unwrapped/package.nix:117]:
```suggestion
    platforms = lib.platforms.unix;
```
`unix` means both `linux` and `darwin`.

**Review by TheMaxMur — APPROVED** [review]

**Pull request closed without merging**